javascript-review.js

```diff
@@ -3,87 +3,227 @@
 // 1.
 
 // Create an object that has four properties. One property should be set so that the value is a number, another property set to a string, the third to an array, and the fourth to a function. This function should simply console.log the value of the first property.
+// let obj = {
+//     num: 1,
+//     str: "my string",
+//     arr: [1, 2, 3],
+//     func: function myfunc() {console.log(this.num)}
+// }
 
 // 2.
 
-// Write a constructor function (to create objects) that takes one argument. It should use that argument to set the value of the first attribute of the object. There should be a second attribute that stores the value ‘red’. 
+// Write a constructor function (to create objects) that takes one argument. It should use that argument to set the value of the first attribute of the object. There should be a second attribute that stores the value ‘red’.
+
+
+// function Square(width) {
+//     this.width = width;
+//     this.color = "red"
+// }
+
+// let square = new Square(50);
+// console.log(square);
+
 
 // Note that a constructor function is the old school syntax that was used in javascript before the sugar class syntax came along. 
 
 // 3.
 
 // Can you access variables defined outside of functions within the scope of functions in JS? Show a simple example of this being used? (Define a constant in global scope and use that in a very simple function.)
 
+// let globalVar = 5;
+
+// The global variable 'globalVar' is accessible by the useGlobalVar function, therefore we can say that
+// it is within the functions scope
+// function useGlobalVar () {
+//     return globalVar;
+// }
+
+// console.log(useGlobalVar());
+
+
 // 4.
 
 // What two ways can be used to access the properties of objects in JS? Give examples of both (although assume the objects have been defined).
+// 1) you can access a property of an object using the dot notation, such as obj.property
+// 2) you can access a property of an object using the bracket notation with the key, such as obj["key"]
+
 
 // 5.
 
 // What is JSON? Be clear but concise.
+// JSON is a data format. It stands for Javascript Object Notation. It is represented as a string object and
+// can be useful as a standard format for transmitting data across a network in a client server model.
+
 
 // 6.
 
 // What is a callback?
+// a callback is a higher order function which is passed in as a parameter to another function to be 
+// 'called back' at a later time. Since it is not evaluated until later, we pass in only the function 
+// definition as a parameter and drop the ending brackets. This is possible because functions are objects
+// in Javascript. 
```

Comment:
awesome answer

i tend to steer clear from saying passed in as a `parameter`, i would use the word `argument`